File: modules/_foundation/src/main/java/com/projecthub/base/shared/domain/event/NotificationEvent.java
package com.projecthub.base.shared.domain.event;

import java.time.Instant;
import java.util.Map;

/**
 * Represents a notification event in the system.
 * 
 * @param recipientId The ID of the user who should receive the notification
 * @param senderId The ID of the user who triggered the notification (optional)
 * @param type The type of notification
 * @param title Short title/summary of the notification
 * @param message Detailed notification message
 * @param metadata Additional contextual data about the notification
 * @param timestamp When the notification was created
 * @param read Whether the notification has been read
 */
public record NotificationEvent(
    String recipientId,
    String senderId,
    NotificationType type,
    String title,
    String message,
    Map<String, Object> metadata,
    Instant timestamp,
    boolean read
) {}

File: src/main/kotlin/com/projecthub/domain/notification/event/NotificationEvent.kt
package com.projecthub.domain.notification.event

import com.projecthub.events.DomainEvent
import java.time.Instant
import java.util.UUID

/**
 * Sealed class representing all possible notification events.
 */
sealed class NotificationEvent : DomainEvent() {
    abstract val notificationId: UUID
    abstract val recipientId: UUID

    data class Created(
        override val notificationId: UUID,
        override val recipientId: UUID,
        val type: String,
        val content: String,
        override val occurredAt: Instant = Instant.now()
    ) : NotificationEvent()

    data class Read(
        override val notificationId: UUID,
        override val recipientId: UUID,
        override val occurredAt: Instant = Instant.now()
    ) : NotificationEvent()

    data class Deleted(
        override val notificationId: UUID,
        override val recipientId: UUID,
        override val occurredAt: Instant = Instant.now()
    ) : NotificationEvent()
}
